base/strings/util.jl

```diff
@@ -57,6 +57,20 @@ function startswith(a::Union{String, SubString{String}},
     end
 end
 
+"""
+    startswith(io::IO, prefix::Union{AbstractString,Base.Chars})
+
+Check if an `IO` object starts with a prefix.
+"""
+function Base.startswith(io::IO, prefix::Union{AbstractString,Base.Chars})
+    pos = position(io)
+    s = _getminprefix(io, prefix)
+    seek(io, pos)
+    return startswith(s, prefix)
+end
+_getminprefix(io::IO, prefix::Union{AbstractString,AbstractChar}) = String(read(io, length(prefix)))
+_getminprefix(io::IO, chars::Union{Tuple{Vararg{<:AbstractChar}},AbstractVector{<:AbstractChar},Set{<:AbstractChar}}) = _getminprefix(io, first(chars))
```

Comment:
These do not correctly handle unicode. If the prefix is a `Chars`, this should read one `Char` from the stream and see if it is in the set.  The prefix should also be converted to `String`, then we can use `ncodeunits` to know how many bytes to read.